storybookjs/storybook
# fix: dont mutate Story prop types to optional

Issue:
When using TypeScript and you specify props, the props were converted to optional. I don't think this is a good pattern. One should type cast if they want to go against specification. I don't see any reason to make it `Partial` in the code base (correct me if I'm wrong, this Partial seems intentional).

## What I did
Removed the `Partial` from the type defs in my node_modules folder and saw proper results. I searched this repo for references to `BaseAnnotations` and found nothing concerning.

## How to test

- Is this testable with Jest or Chromatic screenshots? -> No
- Does this need a new example in the kitchen sink apps? -> No
- Does this need an update to the documentation? -> No
+ I don't know what the reasoning behind this Partial is, I don't see a reason to mutate this the `Args` type to something else. If modules need to alter it, shouldn't they need to be the ones responsible for doing so? Needs to be verified though. This Partial was probably intentional -> Yes

## Example

```
import React from "react";
import { Story } from "@storybook/react";

interface ExampleProps {
  requireMe: "please";
  fine?: true;
}

const Template: Story<ExampleProps> = (args) => <div {...args}>example</div>;

export const Test = Template.bind({});

Test.args = {}; // error requireMe is undefined error should be visible here. This makes it easy to spot mistakes when you make em.
```
